packages/component/lib/element.ts

```diff
@@ -86,6 +87,28 @@ export class JoistElement extends withInjector(HTMLElement) implements Lifecycle
     state.onChange(this.render.bind(this));
   }
 
+  onPropChanges(..._: PropChange[]) {}
+
+  /**
+   * Adds a PropChange to the que.
+   * PropChanges resolves as a micro task once a promise is resolved.
+   * This batches onPropChanges calls
+   */
+  quePropChange(propChange: PropChange) {
+    this.propChangeQue.push(propChange);
+
+    if (!this.markedForCheck) {
+      this.markedForCheck = true;
+
+      Promise.resolve().then(() => {
+        this.onPropChanges(...this.propChangeQue);
+
+        this.markedForCheck = false;
+        this.propChangeQue = [];
+      });
+    }
+  }
```

Comment:
Ultimately what this is saying is. Once a property has been changed a Promise is resolved so that code will be run in a microtask which will be AFTER synchronous code. This means we can add as many PropChanges to the que as we want and will only get a single onPropChanges call.

Reply:
Need to add tests.